src/core/modules/AtmosphereShowMySky.hpp

```diff
@@ -66,7 +66,8 @@ class AtmosphereShowMySky : public Atmosphere, public QObject
 #ifdef ENABLE_SHOWMYSKY
 	QLibrary showMySkyLib;
 	Vec4i viewport;
-	int gridMaxY,gridMaxX;
+	int gridMaxY,gridMaxX,atmoRes;
```

Comment:
I would still like to see some documentation for these variables as long as your memory is fresh! It is much harder to reverse-doc that next year.
@10110111 please also add some class doc and more background info after merge. Consider a new developer wants to start with this in 4 years.

Reply:
Something like that?
```diff
diff --git a/src/core/modules/AtmosphereShowMySky.hpp b/src/core/modules/AtmosphereShowMySky.hpp
index 6eb68ce..fa396fc 100644
--- a/src/core/modules/AtmosphereShowMySky.hpp
+++ b/src/core/modules/AtmosphereShowMySky.hpp
@@ -66,7 +66,17 @@
 #ifdef ENABLE_SHOWMYSKY
 	QLibrary showMySkyLib;
 	Vec4i viewport;
-	int gridMaxY,gridMaxX,atmoRes;
+	int gridMaxY,gridMaxX;
+
+	// atmosphere_resolution_reduction
+	// 1 ~> full resolution (default)
+	// 2 ~> half resolution
+	// 4 ~> quarter resolution
+	int atmoRes;
+
+	// flag_atmosphere_dynamic_resolution
+	// false ~> static resolution (default)
+	// dynamic resolution
 	bool flagDynamicResolution;
 
 	QVector<Vec2f> posGrid;
```

Reply:
Yes, good start! If you mark the comment with exclamation mark like 

//! dynamic resolution reduces resolution of the skylight texture while view is moved. 
//! This increases frame rate esp. on slow systems.

or similar, it will even be included in the Doxygen online docs and describes the motivation for this feature. (Note that the description of the PR and this whole discussion is usually not available while studying the source code at some later date.)

Reply:
so let's try again...
I'm afraid the poor little header might get overloaded with all that stuff. :)
```diff
diff --git a/src/core/modules/AtmosphereShowMySky.hpp b/src/core/modules/AtmosphereShowMySky.hpp
index 6eb68ce..3483efe 100644
--- a/src/core/modules/AtmosphereShowMySky.hpp
+++ b/src/core/modules/AtmosphereShowMySky.hpp
@@ -66,7 +66,33 @@
 #ifdef ENABLE_SHOWMYSKY
 	QLibrary showMySkyLib;
 	Vec4i viewport;
-	int gridMaxY,gridMaxX,atmoRes;
+	int gridMaxY,gridMaxX;
+
+	//! In order to achieve higher frame rates on slow systems, the configuration parameter:
+	//!	config.ini
+	//!	[landscape]
+	//!	atmosphere_resolution_reduction
+	//! allows reducing the resolution of the skylight texture.
+	//! Preferred values are:
+	//!	1 ~> full resolution (default)
+	//!	2 ~> half resolution
+	//!	4 ~> quarter resolution
+	int atmoRes;
+
+	//! The configuration switch:
+	//!	config.ini
+	//!	[landscape]
+	//!	flag_atmosphere_dynamic_resolution
+	//! allows to use the reduced resolution only while moving the view,
+	//! when panning, zooming, dimming or in time-lapse mode.
+	//! With the real-time display, on the other hand, the full resolution is retained.
+	//! Possible values are:
+	//!	false ~> static resolution (default)
+	//!	true ~> dynamic resolution
+	//! A notice:
+	//! In dynamic resolution mode, a motion detector selects either full or reduced resolution.
+	//! The change in resolution could be particularly visible in close proximity to the Sun.
+	//! Especially at full resolution, frames will be skipped depending on the speed of movement.
 	bool flagDynamicResolution;
 
 	QVector<Vec2f> posGrid;
```

Reply:
Good! The header file can sustain such augmentations without great pain.

Reply:
In fact, Doxygen also supports an easier syntax, using `/*! some docs, even multiple lines */`. So no need to do this `//!` on each line.

Reply:
With this syntax everything can be hidden again. A feast for the eyes for code purists. :)

![Screenshot 2022-09-23 145511](https://user-images.githubusercontent.com/112473593/191965752-42d3b3ec-569f-42fa-bd22-7d7e74fcf590.png)

Reply:
@10110111 is technically right, but we have our long-used style. @alex-w ?

Reply:
We used both styles